### PR TITLE
Use new custom raft port support

### PR DESCRIPTION
**Pull request by bts**

### CODE REVIEW

File: src/QuorumTools/Cluster.hs [hunk 1]
```diff
@@ -62,7 +70,7 @@ emptyClusterEnv = ClusterEnv
   , _clusterConstellationConfs    = Map.empty
   , _clusterAccountKeys           = Map.empty
   , _clusterInitialMembers        = Set.empty
-  , _clusterConsensus             = Raft
+  , _clusterConsensus             = Raft 50400
```
Comment (joelburget) [inline src/QuorumTools/Cluster.hs]: Slight preference for `Raft { _raftBasePort = 50400 }`

File: src/QuorumTools/Cluster.hs [hunk 2]
```diff
@@ -102,6 +110,10 @@ httpPort (GethId gid) = (fromIntegral gid +) <$> view clusterBaseHttpPort
 rpcPort :: HasEnv m => GethId -> m Port
 rpcPort (GethId gid) = (fromIntegral gid +) <$> view clusterBaseRpcPort
 
+raftPort :: HasEnv m => GethId -> m (Maybe Port)
+raftPort (GethId gid) = fmap (fromIntegral gid +) . getFirst
```
Comment (joelburget) [inline src/QuorumTools/Cluster.hs]: Slight preference for `do`ing this in two lines.

File: src/QuorumTools/Cluster.hs [hunk 3]
```diff
@@ -316,7 +349,11 @@ createNode genesisJsonPath gid = do
     let acctKey = forceAcctKey $ cEnv ^? clusterAccountKeys . ix gid
     installAccountKey gid acctKey
     eid <- requestEnodeId gid
-    mkGeth gid eid
+    mRaftPort <- raftPort gid
```
Comment (joelburget) [inline src/QuorumTools/Cluster.hs]: The control flow is kind of complicated. Mind explaining it to me (in person is fine)? Here we ask the geth node for its enode id so we can parse out its port. Why don't we have the port already? I'm sure this makes sense but I want to understand all the steps.

Comment (bts) [reply src/QuorumTools/Cluster.hs]: Yeah it's a little weird -- I'll add a comment.  In this scenario, when we ask the geth node for `admin.nodeInfo.enode`, it comes back with a `discport`, but no `raftport`. We are trying to construct an `EnodeId` here that includes a `raftport` querystring parameter. So we get the enode-id-sans-raftport, the port from the `raftPort` function, and build the new EnodeId with `addRaftPort`.

Comment (bts) [reply src/QuorumTools/Cluster.hs]: Actually let's see if we can get `admin.nodeInfo.enode` to return a raftport on the geth side.

Comment (bts) [reply src/QuorumTools/Cluster.hs]: I talked to Joel about this. This is difficult, but will be easier/cleaner to implement once we make `--raft` imply `--nodiscover` on the geth side. Punting for now

**Pull request merged**